### PR TITLE
Added feature to upload store logo from admin

### PR DESCRIPTION
**Pull request by rav-redchamps**
This pull request allows to directly upload an image instead of specifying the path in the text input box at below admin settings

1. System > Configuration > Design > Header > Logo Image
2. System > Configuration > Design > Header > Small Logo Image

### CODE REVIEW

**Comment by sreichel** [comment]
For the copyright changes ... why `@author     Magento Core Team <core@magentocommerce.com>`?

Not blocking, but it seems wrong .... it shpould be "OpenMage" and some email address - is there any?

**Comment by rav-redchamps** [comment]
> For the copyright changes ... why `@author Magento Core Team <core@magentocommerce.com>`?
> 
> Not blocking, but it seems wrong .... it shpould be "OpenMage" and some email address - is there any?

I thought the same. But couldn't find any reference from another core file that should be added there in regards to OpenMage. Almost all files contain Magento copyright heads.

Please let me know what info should be added, I will update it.

**Comment by sreichel** [comment]
> Please let me know what info should be added, I will update it.

No idea. But "Magento Core Team" is just wrong for a new class. I'm sure there are other files added via OpenMage and they all have "Magento Copyrights" ...

**Comment by rav-redchamps** [comment]
Yes definitely "Magento Core Team" is wrong. Just replaced it to "OpenMage" & remove email for now. I can add the email address whenever available.

P.S. I have changed "Magento" name to "OpenMage" & links in the other parts of the doc block as well. There is one email still left "license@magento.com", should I remove it, or do we have a replacement?

**Comment by sreichel** [comment]
> If you did not receive a copy of the license and are unable to
> obtain it through the world-wide-web, please send an email
> to license@magento.com so we can send you a copy immediately.

Imho ... i'd remove it, but i dont know how to deal with legally correct. @Flyingmana @colinmollenhour ?

**Comment by rav-redchamps** [comment]
> Imho ... i'd remove it, but i dont know how to deal with legally correct. @Flyingmana @colinmollenhour ?

In the meantime, removed that part from the doc block. Please feel free to let me know if it needs to be re-added with alterations.

**Comment by Flyingmana** [comment]
just omit the copyright line, or put your own name in there.
Giving the Copyright to us, would require a contributor license agreement and some more legal stuff, but its also not necessary.

**Comment by rav-redchamps** [comment]
> just omit the copyright line, or put your own name in there.
> Giving the Copyright to us, would require a contributor license agreement and some more legal stuff, but its also not necessary.

Okay, replaced it with our name. Please feel free to let me know if any change required.

**Comment by sreichel** [comment]
Should we discuss this? Discord? Internal project?

I dont know how to deal with CLA correctly, but M2 has something in for contributing ...

Adding different agencie/contributor names to new files is no good idea. What if a class is rewritten and majority to changes belong to another author? Imho all new files should have an OpenMage header to keep it trackable.

**Comment by colinmollenhour** [comment]
> Adding different agencie/contributor names to new files is no good idea.

I agree, all files should have a standard header. Is there any value to the `@author` tag? I say just remove it.
I also think the copyright notice should be to OpenMage, but I don't know much about CLAs yet..

**Comment by rav-redchamps** [comment]
Hey Guys,

Anything final on this. Please let me know the final edits that need to be done.

Looking forward to the merging of this pull request.

**Comment by kiatng** [comment]
Is it better to name the class `Mage_Adminhtml_Model_System_Config_Backend_Image_Logo` and place the file in the _Image_ dir; to be consistent with existing classes, eg, `Mage_Adminhtml_Model_System_Config_Backend_Image_Favicon`. 

A small imperfection, existing image sources are incoorectly linked in config:

image

**Comment by rav-redchamps** [comment]
Hi @kiatng 

- Moved backend model to image directory
- Moved logo upload src from the skin to media
- Added OpenMage logo as default logo instead of Magento
- Bugfix: Logo setting value was getting removed if you save config without any change

**Comment by kiatng** [comment]
@rav-redchamps Thanks for your effort. Just reading the code, without any testing, here are some of my concerns but it may be nothing:

1. The core has very strick separation between frontend and backend, this breaks the separation: `$folderName = Mage_Adminhtml_Model_System_Config_Backend_Image_Logo::UPLOAD_DIR;`.
2. `<logo_alt>Magento Commerce</logo_alt>` should be `<logo_alt>OpenMage</logo_alt>`.
3. While I think uploading to the `media` dir is good and consistent with existing uploads from users, it may create backward compatibility issue. 

Let's slow down and be a bit more cautious.

**Comment by fballiano** [comment]
I'm thinking, nowadays there are probably multiple logos (2x resolutions etc) so... is uploading logo/small_logo the modern solution we need?
I really don't have an answer but I think it could be something to think about

File: app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Image/Logo.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade OpenMage to newer
+ * versions in the future. If you wish to customize OpenMage for your
+ * needs please refer to https://www.openmage.org for more information.
+ *
+ * @category    Mage
+ * @package     Mage_Adminhtml
+ * @copyright  Copyright (c) RedChamps. (https://redchamps.com)
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Model/System/Config/Backend/Image/Logo.php:20]: this was already commented in the issue but I really think it's a bad idea to have another copyright holder in our codebase

**Comment by fballiano** [comment]
closing since we've no feedback from @rav-redchamps since more than a year, feel free to reopen in case there's news

**Pull request closed without merging**